typing/includemod.ml

```diff
@@ -406,6 +406,19 @@ module Sign_diff = struct
     }
 end
 
+type 'a core_incl0 = 'a -> 'a -> (module_coercion, Error.sigitem_symptom) result
+type 'a core_incl =
```

Comment:
I would expect a comment that explains what this is about.

(My understanding: we can instantiate the module-level subtyping check with different notion of what it means for "core constructs" (term-level constructs, not module-level constructs) to be compatible, and in particular one instance checks full subtyping and another instance just check a form of compatibility of arities to guarantee well-formedness.)